src/sidebar/components/filter-status.js

```diff
@@ -0,0 +1,156 @@
+ *   the selection. This mode supersedes all others (i.e. other filters ignored).
+ * - query: There is a user-entered filter query. It can "stack" with user-focus.
+ *    Button clears the selection (query, etc.), but will leave user-focus mode
+ *    intact.
+ * - focus: Neither of the above two modes apply and user-focus mode is
+ *   configured (it may or may not be active). Button toggles activation of
+ *   user-focus mode.
+ *
+ * Descriptive text about the filter status and annotations follows the
+ * pattern:
+ *
+ *   [Showing] (<resultCount>|No) (annotation[s]|result[s]) [for <filterQuery>]
+ *   [by <focusedUser>] [\(and <forcedCount> more\)]`
+ */
+export default function FilterStatus() {
+  const rootThread = useRootThread();
+
+  const filterState = useStore(store => store.filterState());
+
+  // The total count of all of the annotations in the store—may differ from
+  // the number of annotations in the thread and the number of visible annotations
+  const totalCount = useStore(store => store.annotationCount());
+
+  // Actions
+  const clearSelection = useStore(store => store.clearSelection);
+  const toggleFocusMode = useStore(store => store.toggleFocusMode);
+
+  const filterMode = (() => {
+    if (filterState.selectedCount > 0) {
+      return 'selection';
+    } else if (filterState.filterQuery) {
+      return 'query';
+    } else if (filterState.focusConfigured) {
+      return 'focus';
+    }
+    return null;
+  })();
+
+  if (!filterMode) {
+    // Nothing to do here
+    return null;
+  }
+
+  // Some threads in the `visibleCount` may have been "forced visible" by
+  // the user by clicking "Show x more in conversation" — subtract these
+  // forced-visible threads out to get a correct count of actual filter matches.
+  // In 'selection' mode, rely on the count of selected annotations
+  const visibleCount = countVisible(rootThread);
+  const resultCount =
+    filterMode === 'selection'
+      ? filterState.selectedCount
+      : visibleCount - filterState.forcedVisibleCount;
+
+  let buttonText;
+  switch (filterMode) {
+    case 'selection':
+      // In user-focus mode, displayed counts include annotations AND replies,
+      // while `totalCount` only includes top-level annotations, so that count
+      // doesn't make sense in user-focus mode
+      buttonText = filterState.focusConfigured
+        ? 'Show all'
+        : `Show all (${totalCount})`;
+      break;
+    case 'focus':
+      if (!filterState.forcedVisibleCount) {
+        buttonText = filterState.focusActive
+          ? 'Show all'
+          : `Show only ${filterState.focusDisplayName}`;
+      } else {
+        // When user focus is applied and there are some forced-visible threads,
+        // this special case applies. The button will clear force-visible threads
+        // but leave user focus intact
+        buttonText = 'Reset filters';
+      }
+      break;
+    case 'query':
+    default:
```

Comment:
If `filterMode` has no possibilities other than `selection`, `focus` or `query` then you don't need the `default` case here.